code/modules/mob/living/simple_animal/morph/morph.dm

```diff
@@ -0,0 +1,179 @@
+		return 0
+	return 1
+
+/mob/living/simple_animal/hostile/morph/proc/eat(atom/movable/A)
+	if(A && A.loc != src)
+		visible_message("<span class='warning'>[src] swallows [A] whole!</span>")
+		A.forceMove(src)
+		return 1
+	return 0
+
+/mob/living/simple_animal/hostile/morph/ShiftClickOn(atom/movable/A)
+	if(morph_time <= world.time && !stat)
+		if(A == src)
+			restore()
+			return
+		if(istype(A) && allowed(A))
+			assume(A)
+	else
+		src << "<span class='warning'>Your chameleon skin is still repairing itself!</span>"
+		..()
+
+/mob/living/simple_animal/hostile/morph/proc/assume(atom/movable/target)
+	morphed = 1
+	form = target
+
+	appearance = target.appearance
+	transform = initial(transform)
+	pixel_y = initial(pixel_y)
+	pixel_x = initial(pixel_x)
+	visible_message("<span class='warning'>[src] suddenly twists and changes shape, becoming a copy of [target]!</span>", \
+					"<span class='notice'>You twist your body and assume the form of [target].</span>")
+	//Morphed is weaker
+	melee_damage_lower = 5
+	melee_damage_upper = 5
+	speed = 0
+
+	morph_time = world.time + MORPH_COOLDOWN
+	return
+
+/mob/living/simple_animal/hostile/morph/proc/restore()
+	if(!morphed)
+		return
+	morphed = 0
+	form = null
+
+	//anim(loc,src,'icons/mob/mob.dmi',,"morph",,src.dir)
+
+	visible_message("<span class='warning'>[src] suddenly collapses in on itself, dissolving into a pile of green flesh!</span>", \
+					"<span class='notice'>You reform to your normal body.</span>")
+	name = initial(name)
+	icon = initial(icon)
+	icon_state = initial(icon_state)
+	overlays.Cut()
+
+	//Baseline stats
+	melee_damage_lower = initial(melee_damage_lower)
+	melee_damage_upper = initial(melee_damage_upper)
+	speed = initial(speed)
+
+	morph_time = world.time + MORPH_COOLDOWN
+
+/mob/living/simple_animal/hostile/morph/handle_state_icons()
+	return
+
+/mob/living/simple_animal/hostile/morph/death(gibbed)
+	if(morphed)
+		visible_message("<span class='warning'>[src] twists and dissolves into a pile of green flesh!</span>", \
+						"<span class='userdanger'>Your skin ruptures! Your flesh breaks apart! No disguise can ward off de--</span>")
+		restore()
+	if(gibbed)
+		for(var/atom/movable/AM in src)
+			AM.forceMove(loc)
+			if(prob(90))
+				step(AM, pick(alldirs))
+	..(gibbed)
+
+/mob/living/simple_animal/hostile/morph/Aggro() // automated only
+	..()
+	restore()
+
+/mob/living/simple_animal/hostile/morph/LoseAggro()
+	vision_range = idle_vision_range
+
+/mob/living/simple_animal/hostile/morph/AIShouldSleep(var/list/possible_targets)
+	. = ..()
+	if(.)
+		var/list/things = list()
+		for(var/atom/movable/A in view(src))
+			if(allowed(A))
+				things += A
+		var/atom/movable/T = pick(things)
+		assume(T)
+
+/mob/living/simple_animal/hostile/morph/AttackingTarget()
+	if(isliving(target)) // Eat Corpses to regen health
+		var/mob/living/L = target
+		if(L.stat == DEAD)
+			if(do_after(src, 30, target = L))
+				if(eat(L))
+					heal_overall_damage(150,150)
```

Comment:
I think this only heals 50 damage on TG? Not sure...they use a different proc, too.